test/common/router/retry_state_impl_test.cc

```diff
@@ -412,6 +412,65 @@ TEST_F(RouterRetryStateImplTest, RetriableStatusCodesHeader) {
   }
 }
 
+TEST_F(RouterRetryStateImplTest, RetriableHeadersPolicy) {
```

Comment:
Please add a short description of what each test is checking for (here and for the other TEST_F declarations).

Also, should there be a test where `policy_.retry_on_` is set to "retriable-headers"?

Reply:
Done.